File: level3/GenomicRangeQuery.js
function solution(S, P, Q) {
    var v = {'A':1,'C':2,'G':3,'T':4};
    var s = S.split("").map(function(e){return v[e];});
    var result = [];
    var qi = 0;
    for (var i=0; i<P.length; i++){
        qi = (Q[i]==P[i])? Q[i]+1: Q[i];
        result[i] = Math.min.apply(null, s.slice(P[i],qi));
    }
    return result;
}

File: python_solution/level3/genomicRangeQuery.py
def solution(S, P, Q):
    # write your code in Python 2.7
    impact = {'A':1, 'C':2, 'G':3, 'T':4}
    s = map(lambda x: impact[x] , S)
    result = []    
    for i in range(0,len(P)):
        i1 = P[i]
        i2 = Q[i] if P[i] != Q[i] else Q[i] + 1
        result.append(min(s[i1:i2]))
    return result
    pass
